implementations/rust/ockam/ockam_transport_core/src/tcp/workers/receiver.rs

```diff
@@ -1,51 +1,53 @@
-use ockam_core::async_trait;
-use ockam_core::{Address, Decodable, LocalMessage, Processor, Result, TransportMessage};
+use crate::TransportError;
+use ockam_core::compat::{boxed::Box, vec::Vec};
+use ockam_core::{
+    async_trait, Address, Decodable, LocalMessage, Processor, Result, TransportMessage,
+};
 use ockam_node::Context;
-use ockam_transport_core::TransportError;
-use tokio::{io::AsyncReadExt, net::tcp::OwnedReadHalf};
 use tracing::{error, info, trace};
 
+use crate::tcp::traits::io::{AsyncRead, AsyncReadExt};
+
 /// A TCP receiving message worker
 ///
-/// Create this worker type by calling
-/// [`start_tcp_worker`](crate::start_tcp_worker)!
+/// This type will be created by [TcpSendWorker::start_pair](super::sender::TcpSendWorker::start_pair)
 ///
 /// This half of the worker is created when spawning a new connection
 /// worker pair, and listens for incoming TCP packets, to relay into
 /// the node message system.
-pub(crate) struct TcpRecvProcessor {
-    rx: OwnedReadHalf,
+pub(crate) struct TcpRecvProcessor<R> {
     peer_addr: Address,
+    tcp_stream: R,
+    cluster_name: &'static str,
```

Comment:
Is there a reason to store the `cluster_name` again here?

Reply:
It was a vestigial change when I took the `cluster_name` as a parameter from outside the crate and I needed to store it to use it in the worker's `initialize`. But since I [changed it back](https://github.com/ockam-network/ockam/pull/2213#discussion_r812353796) to taking the `cluster_name` from outside I need to keep it.